apps/platform/src/components/dashboard/secret/decryptSecret/index.tsx

```diff
@@ -0,0 +1,61 @@
+'use client'
+import { useAtom, useAtomValue } from 'jotai'
+import { useMemo } from 'react'
+import { InfoSVG } from '@public/svg/shared'
+import { revealSecretOpenAtom, selectedProjectAtom } from '@/store'
+import { Switch } from '@/components/ui/switch'
+import {
+  Tooltip,
+  TooltipArrow,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger
+} from '@/components/ui/tooltip'
+
+export default function DecryptSecret(): React.JSX.Element {
+  const [revealSecretOpen, setRevealSecretOpen] = useAtom(revealSecretOpenAtom)
+  const selectedProject = useAtomValue(selectedProjectAtom)
+
+  const isPrivateKeyStored = useMemo(() => {
+    return (
+      selectedProject?.storePrivateKey === true ||
+      Boolean(localStorage.getItem(`${selectedProject?.name}_pk`))
```

Comment:
You can remove this line. This toggle can only be set if private key is stored in the db.